### ISSUE TITLE
Local LLM URL Fails

### ISSUE THREAD

**Issue by danielsawan** [sashirestela/simple-openai#198]
Hello,

I am using the SimpleOpenAI library with the following configuration:

```java
SimpleOpenAI openAI = SimpleOpenAI.builder()
//.baseUrl("https://api.openai.com/") // works correctly
.baseUrl("http://192.168.1.21:1234/") // does not work
.apiKey("sk-xxxxx")
.build();
```

When I use the official OpenAI API URL (https://api.openai.com/), everything works perfectly. However, I have a local LLM that is compatible with the OpenAI API and works fine when I make requests via the curl command. But when I configure the local URL in the code (http://192.168.1.21:1234/), it doesn't work, and no exception is thrown to indicate the issue.

I'm trying to understand why the local URL doesn't work in this configuration, despite working correctly with curl, and why no exception is generated.

Thank you in advance for your help.

**Comment by sashirestela** [comment]
Hi @danielsawan, thanks for using simple-openai!

It could be an issue with your firewall. Take a look at a similar issue here: https://github.com/sashirestela/simple-openai/issues/172

**Comment by danielsawan** [comment]
Hi @sashirestela,

Thank you for your response! It doesn’t seem likely to me that the issue is caused by a firewall, as the curl call to the URL from the same host running this Java code works perfectly. This indicates that the machine hosting the code can indeed reach the local URL.

I’ll take a closer look at this tomorrow, but I doubt that the firewall is the cause.

Otherwise, should this code work in this case? I mean, is it possible to connect your library to a locally hosted LLM via LLM Studio?

Thank you in advance for your insights!

**Comment by sashirestela** [comment]
> Otherwise, should this code work in this case? I mean, is it possible to connect your library to a locally hosted LLM via LLM Studio?
I really don't know. I cannot install LLM Studio due to local restrictions.
Maybe you could ask the author of the #172 issue.

**Comment by sashirestela** [comment]
This issue is open for anyone to make a research.

**Comment by danielsawan** [comment]
Hello @sashirestela,

After some further testing, I found a solution that works for me. I was able to connect to LM Studio using a custom `HttpClient`. Here is the code that solved my problem:

````java
HttpClient httpClient = HttpClient.newBuilder()
    .version(HttpClient.Version.HTTP_1_1)
    .followRedirects(HttpClient.Redirect.NORMAL)
    .connectTimeout(Duration.ofSeconds(20))
    .build();

openAI = SimpleOpenAI.builder()
    .baseUrl("http://0.0.0.0:1234") // LM Studio
    .apiKey("sk-****")
    .httpClient(httpClient)
    .build();
````
By adding my own HttpClient, I was able to establish a connection with LM Studio without any issues. It is possible that the default HttpClient used by the library does not handle certain aspects specific to communications with LM Studio.

I hope this can help others who have encountered similar issues.

Thanks again for your assistance!
